packages/bvaughn-architecture-demo/src/suspense/PauseCache.ts

```diff
@@ -172,3 +176,25 @@ async function fetchPauseId(
     wakeable.reject(error);
   }
 }
+
+// TODO Hook into e2e tests to assert 0-1 pause id per point
+export function trackExecutionPointPauseIds(
+  executionPoint: ExecutionPoint,
+  pauseId: PauseId
+): void {
+  const set = executionPointToPauseIdSetMap.get(executionPoint);
+  if (set == null) {
+    executionPointToPauseIdSetMap.set(executionPoint, new Set([pauseId]));
+  } else {
+    set.add(pauseId);
+
+    const pauseIds = Array.from(set).join(",");
+    const error = new Error(
+      `Duplicate pause ids found for point "${executionPoint}" (${pauseIds})`
+    );
+
+    captureException(error);
```

Comment:
I'm not very familiar with Sentry so this might be the wrong way to approach this problem. Feedback welcome!